packages/opencensus-exporter-trace-zipkin/test/test-zipkin.ts

```diff
@@ -0,0 +1,145 @@
+  url: 'http://localhost:9411/api/v2/spans',
+  serviceName: 'opencensus-tests'
+} as ZipkinExporterOptions;
+
+/** Default config for traces tests */
+const defaultConfig: types.TracerConfig = {
+  samplingRate: 1
+};
+
+/** Run a nock server to replace zipkin service */
+const runNockServer = () => {
+  nock('http://localhost:9411/')
+      .persist()
+      .post('/api/v2/spans')
+      .reply(202)
+      .post('/api/v2/spanswrong')
+      .reply(404);
+};
+
+/**
+ * Checks if zipkin service is not running for tests
+ */
+const isNotZipkinRunning = (callback: Function) => {
+  http.get('http://localhost:9411', (res) => {
+        if (res.statusCode !== 200) {
+          callback();
+        }
+      }).on('error', (e) => {
+    callback();
+  });
+};
+
+/** Checking if zipkin service is running, otherwise run a nock server */
+before((done) => {
+  isNotZipkinRunning(() => {
+    runNockServer();
+    done();
+  });
+});
+
+/** Zipkin tests */
+describe('Zipkin Exporter', function() {
+  /** Desabling the timeout for tests */
+  this.timeout(0);
+  /** Should create a Zipkin instance */
+  describe('new Zipkin()', () => {
+    it('should create a Zipkin instance', () => {
+      const zipkin = new ZipkinTraceExporter(zipkinOptions);
+      assert.ok(zipkin instanceof ZipkinTraceExporter);
+    });
+  });
+
+  /** Should called when a rootSpan ended */
+  describe('onEndSpan()', () => {
+    it('Should called when a rootSpan ended', () => {
+      const exporter = new ZipkinTraceExporter(zipkinOptions);
+      const tracer = new classes.Tracer();
+      tracer.registerEndSpanListener(exporter);
+      tracer.start(defaultConfig);
+
+      tracer.startRootSpan({name: 'root-test'}, (rootSpan: types.RootSpan) => {
+        const span = rootSpan.startChildSpan('spanTest', 'spanType');
+        span.end();
+        rootSpan.end();
+        assert.ok(exporter.buffer.getQueue().length > 0);
+      });
+    });
+  });
+
+  /** Should send traces to Zipkin service */
+  describe('publish()', () => {
+    it('should send traces to Zipkin service', () => {
+      const exporter = new ZipkinTraceExporter(zipkinOptions);
+      const tracer = new classes.Tracer();
+      tracer.start(defaultConfig);
+
+      return tracer.startRootSpan(
+          {name: 'root-test'}, async (rootSpan: types.RootSpan) => {
+            const span = rootSpan.startChildSpan('spanTest', 'spanType');
+            span.end();
+            rootSpan.end();
+            return exporter.publish([rootSpan]).then((result) => {
+              const resultObject = result as RequestResponse;
+              assert.equal(resultObject.statusCode, 202);
+            });
+          });
+    });
+  });
+
+  /** Should send traces to Zipkin service */
+  describe('publish() with a wrong Zipkin url', () => {
+    it('shouldn\'t send traces to Zipkin service and return an 404 error',
+       () => {
+         /** adding a string to get a wrong url */
+         zipkinOptions.url += 'wrong';
```

Comment:
Prefer cloning over mutating `zipkinOptions`. Otherwise this test will have an impact on the next test (if there is one).

Reply:
Done.